roles/ceph-config/tasks/main.yml

```diff
@@ -9,6 +9,27 @@
       group: "ceph"
       mode: "0755"
 
+  - name: count number of osds for non-lvm scenario
+    set_fact:
+      num_osds: "{{ devices | length }}"
+    when:
+      - devices|length > 0
+      - (osd_scenario == 'collocated' or osd_scenario == 'non-collocated')
+
+  - name: count number of osds for lvm scenario
+    set_fact:
+      num_osds: "{{ lvm_volumes | length }}"
+    when:
+      - lvm_volumes|length > 0
+      - osd_scenario == 'lvm'
+
+  - name: count number of osds for lvm-batch scenario
+    set_fact:
+      num_osds: "{{ (ceph-volume lvm batch --report devices | from_json()).osds | length }}"
```

Comment:
When https://github.com/ceph/ceph-ansible/pull/3111 merges this command will also need the ``--osds-per-device osds_per_device`` flag added.

Reply:
You're also missing ``--format=json`` here.